core/thread/test/testInterpreterLock.cxx

```diff
@@ -0,0 +1,149 @@
+
+   #include "ROOT/TThreadExecutor.hxx"
+   #include "TVirtualRWMutex.h"
+   #include "TInterpreter.h"
+   #include "TROOT.h"
+   #include <iostream>
+   #include <string>
+   #include <vector>
+   #include <thread>
+
+   #include "gtest/gtest.h"
+
+   static auto gDeclarator =  gInterpreter->Declare("int f(int n) { int x = 0; for (auto i = 0u; i < n; ++i) ++x; return x; }");
+
+   constexpr bool gDebugOrder = false;
+
+   TEST(InterpreterLock, ConcurrentCalc)
+   {
+      bool flag = true;
+      auto func = [&](int) {
+         gInterpreter->Calc("f(10)");
+         R__LOCKGUARD(gROOTMutex);
+         if (gDebugOrder)
+            std::cerr << std::this_thread::get_id() << " : start" << std::endl;
+
+         EXPECT_TRUE(flag) << "Broken flag update on thread " << std::this_thread::get_id();
+         flag = false;
+         gInterpreter->Calc("f(100000)");
+         if (gDebugOrder)
+            std::cerr << std::this_thread::get_id() << " : end" << std::endl;
+         flag = true;
+      };
+      ROOT::TThreadExecutor pool(32);
+      std::vector<int> id;
+      for (auto i = 0u; i < 16; i++) id.push_back(i);
+      pool.Foreach(func, id);
+   }
```

Comment:
we know this only breaks every once in a while, can we run this logic e.g. 100 times in a loop?

Reply:
@oshadura Is there a way to have ctest pass ```--gtest_reapeat=100``` to the test executable?

Reply:
Sorry I didn't see this before...There are two approaches:

```
$ foo_test --gtest_repeat=1000
Repeat foo_test 1000 times and don't stop at failures.

$ foo_test --gtest_repeat=1000 --gtest_break_on_failure
Repeat foo_test 1000 times, stopping at the first failure.  This
is especially useful when running under a debugger: when the test
fails, it will drop into the debugger and you can then inspect
variables and stacks.
```
Which do you prefer? For example it will be easy just adding here argument `REPEAT` https://github.com/root-project/root/blob/master/cmake/modules/RootMacros.cmake#L1685

Reply:
In this case `--gtest_repeat=1000 --gtest_break_on_failure` but we probably need to allow for customizing the number of repetition.

Reply:
in this particular case can't we just put a literal for loop around the test body?

Reply:
I find it more expressive and vaguely more practical when debugging to use the gtest feature (for better or worse it also just repeat all the tests :) ).